lib/src/parser.leg

```diff
@@ -84,6 +84,11 @@ schema-command =
     ) ~{ERR("a schema command")}
     (SEMICOLON | EOF)
 
+named-path-clause =
+    < PATHPATTERN i:identifier EQUAL - p:pattern-path -
+    (WHERE e:expression | e:_null_) 
+    >                                   { $$ = named_path_predicate(i, p, e); }
```

Comment:
Here we use named_path_predicate (almost the same as named_path) for named path patterns construction. So the node of type CYPHER_AST_NAMED_PATH may appear both as child of pattern and as child of query. Named path patterns and named paths are structurally identical but different in semantics. So should we create new concrete AST node for named-path-clause in this case?